frontend/lib/src/components/elements/Dialog/Dialog.tsx

```diff
@@ -54,15 +54,21 @@ const Dialog: React.FC<React.PropsWithChildren<Props>> = ({
   element,
   children,
 }): ReactElement => {
-  const { title, dismissible, width, isOpen: initialIsOpen } = element
+  const {
+    title,
+    dismissible,
+    width,
+    isOpen: initialIsOpen,
+    updateId,
+  } = element
   const [isOpen, setIsOpen] = useState<boolean>(false)
 
   useEffect(() => {
     // Only apply the open state if it was actually set in the proto.
     if (notNullOrUndefined(initialIsOpen)) {
       setIsOpen(initialIsOpen)
     }
-  }, [initialIsOpen])
+  }, [initialIsOpen, updateId])
```

Comment:
A bit of an assumption: Potentially, something like this could also solve it in a bit more generic way:

```typescript
  useEffect(() => {
    // Only apply the open state if it was actually set in the proto.
    if (notNullOrUndefined(element.isOpen)) {
      setIsOpen(element.isOpen)
    }
  }, [element])
```

This might work because the `element` dependency is checked via shallow reference equality checks. The element reference will stay the same during renderings as long as no new element message is received. This effect will run whenever a new element message comes in (for every rerun and update messages).